CONTRIBUTING.md

```diff
@@ -330,6 +331,50 @@ internals::impl_from_infallible!(Error);
 All errors that live in an `error` module (eg, `foo/error.rs`) and appear in a public function in
 `foo` module should be available from `foo` i.e., should be re-exported from `foo/mod.rs`.
 
+##### `expect` messages
+
+With respect to `expect` messages, they should follow the
+[Rust standard library guidelines](https://doc.rust-lang.org/std/option/enum.Option.html#recommended-message-style).
+More specifically, `expect` messages should be used to to describe the reason
+you expect the operation to succeed.
+For example, this `expect` message clearly states why the operation should succeed:
+
+```rust
+/// Serializes the public key to bytes.
+pub fn to_bytes(self) -> Vec<u8> {
+    let mut buf = Vec::new();
+    self.write_into(&mut buf).expect("vecs don't error");
+    buf
+}
+```
+
+Also note that `expect` messages, as with all error messages, should be lower
+lower case, except for proper nouns and variable names.
+
+The details on why we chose this style are in the following collapsible section:
+
+<details>
```

Comment:
You can add `<summary>` tag to make it a bit nicer. E.g.

```markdown
<details>
<summary>The details on why we chose this style</summary>

According to the [Rust standard library](https://doc.rust-lang.org/std/error/index.html#common-message-styles),
there are two common styles for how to write `expect` messages:

...
</details>
```

Becomes:

<details>
<summary>The details on why we chose this style</summary>

According to the [Rust standard library](https://doc.rust-lang.org/std/error/index.html#common-message-styles),
there are two common styles for how to write `expect` messages:

...
</details>

Reply:
Great! Done!